packages/cli/src/cli/config/action.import.ts

```diff
@@ -73,16 +82,29 @@ export class CommandImport extends CommandLineAction {
       if (!res.ok) throw new Error('Basemaps is unhealthy');
     }
 
+    if (backup) {
+      await fsa.writeJson(backup, this.backupConfig.toJson());
+    }
+
     if (commit !== true) logger.info('DryRun:Done');
   }
 
   update(config: BaseConfig, commit: boolean): void {
     const promise = Q(async (): Promise<boolean> => {
       const updater = new Updater(config, commit);
+      const db = updater.getDB();
+
       const hasChanges = await updater.reconcile();
-      if (hasChanges) this.invalidations.push(updater.invalidatePath());
+      if (hasChanges) {
+        this.invalidations.push(updater.invalidatePath());
+        const oldData = await db.get(config.id);
+        if (oldData != null) this.backupConfig.put(oldData); // No need to backup anything if there is new insert
+      } else {
+        this.backupConfig.put(config);
```

Comment:
we only need to store the changed data.

Reply:
I am thinking to store the full config data which will same as the config json as this is not very big and also this make import and re-use much easier.